### PR TITLE
Decoupling: Help Center

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

Since there are some WordPress-specific items in quick tips, they need to be moved to `wp-story-editor`

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses https://github.com/google/web-stories-wp/issues/2918

### CODE REVIEW

File: packages/story-editor/src/components/helpCenter/constants.js
Comment (swissspidy) [inline packages/story-editor/src/components/helpCenter/constants.js:173]: After reading the current code on `main` I really don't understand why `TIPS_KEY_MAP` is needed.

Everywhere it's just really used with `Object.keys(TIP_KEYS_MAP)`. This could just be replaced with `Object.keys(TIPS)` or `TIPS` (or well, `TIPS + additionalTips`) directly.

File: packages/story-editor/src/app/helpCenter/provider.js
Comment (maxyinger) [inline packages/story-editor/src/app/helpCenter/provider.js:114]: 👍

File: packages/story-editor/src/app/helpCenter/useHelpCenter/effects.js
Comment (maxyinger) [inline packages/story-editor/src/app/helpCenter/useHelpCenter/effects.js]: I think the expected behavior could be documented a little better here, sorry.

So one of the UX features we have to accommodate here is appending all the unread tips to the end so a user can click into any tip, then press `next` all the way until they get to the last tip and have viewed all unread tips.

ie:
- lets say tips are [tip1, tip2, tip3, tip4] and they're all unread
- user opens the main menu
- user clicks into tip3 first
- the resulting navigation flow should look like: [tip1, tip2, tip3, tip4, tip1, tip2]

I think you can just update this to be the following and it should be fine:
 ```
   const baseNavigationFlow = next.tipKeys;
   const appendedTips = baseNavigationFlow;
    // Tips before current index
    .slice(0, next.navigationIndex)
    // Filter out read tips
    .filter((tip) => !next.readTips[tip]);
  return {
    navigationFlow: [
      ...baseNavigationFlow,
      ...appendedTips,
      DONE_TIP_ENTRY[0],
    ],
  };
 ```

File: packages/story-editor/src/components/helpCenter/constants.js
Comment (maxyinger) [reply packages/story-editor/src/components/helpCenter/constants.js:173]: yea I think the original intention was to keep a consistent type with the read tips of `{ [k in keyof tips]: boolean }`, but I think you're right and you could substitute out `TIPS + additionalTips` it would work the same

File: packages/story-editor/src/components/helpCenter/menu/tips.js
Comment (maxyinger) [inline packages/story-editor/src/components/helpCenter/menu/tips.js]: do we want to memoize this at all just to prevent any re-renders?

I'd also be open to camel casing it since it's not a module scoped constant

File: packages/story-editor/src/app/helpCenter/useHelpCenter/effects.js
Comment (maxyinger) [reply packages/story-editor/src/app/helpCenter/useHelpCenter/effects.js]: you can test this behavior out manually by clearing all your read tips as a user by typing this into the console to clear all your read tips on the backend:
```
await wp.apiFetch({ path: '/web-stories/v1/users/me', method: 'POST', data: { meta: { web_stories_onboarding: {} }}})
```

**Review by maxyinger — COMMENTED** [review]
Awesome work! Looks great so far. Let me know if you got any Q's along the way

File: packages/story-editor/src/components/helpCenter/constants.js
Comment (mohdsayed) [reply packages/story-editor/src/components/helpCenter/constants.js:173]: Removed in https://github.com/google/web-stories-wp/pull/9825/commits/edf162cfd348f117591a13adac662541b54198e1

File: packages/story-editor/src/app/helpCenter/useHelpCenter/effects.js
Comment (mohdsayed) [reply packages/story-editor/src/app/helpCenter/useHelpCenter/effects.js]: Thanks updated

File: packages/story-editor/src/components/helpCenter/menu/tips.js
Comment (mohdsayed) [reply packages/story-editor/src/components/helpCenter/menu/tips.js]: Thanks, yes it should have been camel case, updated. 

Curious to know how memoizing it will prevent re-renders?

**Review by swissspidy — APPROVED** [review]
LGTM; just waiting for the others' approval

**Review by BrittanyIRL — APPROVED** [review]
This looks great and works as expected in the editor, updates make sense to me.

File: packages/story-editor/src/components/helpCenter/menu/tips.js
Comment (maxyinger) [reply packages/story-editor/src/components/helpCenter/menu/tips.js]: Oh yea I guess you're right. It won't. never mind

**Review by maxyinger — APPROVED** [review]

**Pull request merged**
